### DISCUSSION TITLE
ITEAD Sonoff ZigBee 3.0 USB Dongle Plus v2 zigbee2mqtt

### DISCUSSION THREAD

**Discussion by Eros-ITA** [indomus/forum#1909] [HUB: Home Assistant]
Ciao,
sono ormai due giorni che impazzisco con questa chiavetta.

l'ho inserita nel rpi3 e pare l'abbia correttamente riconosciuta

<img width="552" alt="Schermata 2022-09-11 alle 15 54 13" src="https://user-images.githubusercontent.com/94871490/189531638-a1d03b28-6c92-4cce-8466-f038ac4f6c6a.png">

ho seguito tutte le guide per installare MQTT il broker e zigbee2mqtt ma non funziona mi da sempre questo errore 

<img width="793" alt="Schermata 2022-09-11 alle 15 56 47" src="https://user-images.githubusercontent.com/94871490/189531702-d6d42607-6bd9-4567-8d9e-33b09add121e.png">

il dispositivo che viene riconosciuto in automatico l'ho disattivato per evitare conflitti 

questo è lo yaml di configurazione

<img width="743" alt="Schermata 2022-09-11 alle 16 03 58" src="https://user-images.githubusercontent.com/94871490/189531802-4877f0c5-feaf-475d-b1a9-42aea5d06333.png">

grazie a chiunque provi ad aiutarmi

**Comment by snakuzzo** [comment]
Sei sicuro che la porta `/dev/ttyACM0` sia corretta?

**Reply by Eros-ITA** [reply]
> Sei sicuro che la porta `/dev/ttyACM0`sia corretta?

è quello che ho letto qui, è corretto ?

<img width="552" alt="sonoff" src="https://user-images.githubusercontent.com/94871490/189537123-8f62c14f-876a-4959-9cdf-860a230c9027.png">

**Reply by snakuzzo** [reply]
Questo da dove lo vedi?
Comunque ad occhio direi di sì, ma...
hai per caso flashato la chiavetta ? Sicuro di aver messo il firmware giusto ([QUESTO](https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/tree/master/Dongle-E/NCP)) ?

**Reply by Eros-ITA** [reply]
La chiavetta nn l'ho flashata, aperta dalla confezione e inserita nel Raspberry ma mi lo stesso problema anche sul qnap.
La schermata arriva da impostazioni/sistema/hardware 3puntini tutto l'hardware

L'unica cosa che mi fa strano è che ho lo stesso deivename sia su qnap che su Raspberry

**Reply by snakuzzo** [reply]
Non saprei che dirti...prova a flasharla seguendo le istruzioni di itead

**Reply by Eros-ITA** [reply]
forse sono riuscito a farla funzionare, bisogna mettere la prima riga sotto DEVLINKS come port:
e aggiungere adapter: ezsp

per intenderci nel mio caso è cosi:
port: >-
  /dev/serial/by-id/usb-ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20220707170845-if00
adapter: ezsp

spero di esser stato utile a qualcuno

**Reply by snakuzzo** [reply]
Ah giusto...il tipo di `adapter`

**Reply by Eros-ITA** [reply]
cmq continua a non trovarmi nessun dispositivo quindi direi che ho risolto poco o niente

**Reply by KC-inDomus** [reply]
> forse sono riuscito a farla funzionare, bisogna mettere la prima riga sotto DEVLINKS come port: e aggiungere adapter: ezsp
> 
> per intenderci nel mio caso è cosi: port: >- /dev/serial/by-id/usb-ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20220707170845-if00 adapter: ezsp
> 
> spero di esser stato utile a qualcuno

grazie, messo in guida. 👍

**Reply by xCapeITA** [reply]
> cmq continua a non trovarmi nessun dispositivo quindi direi che ho risolto poco o niente

Anche io sto litigando con questa chiavetta dopo averla presa vedendo il post su indomus.
Uso Z2M e a volte pare che la chiavetta venga riconosciuta, a volte no.
Come scritto dall'autore di Z2M tutti i chip EFRxxx sono in via sperimentale e il supporto è inadeguato.
Viene consigliato di
1) cambiare FW del coordinator (che non trovo. trovo solo quelli per la v1 con chip CPxxxx)
2) Di usare Z2M in versione EDGE (developer)
3) di pregare

per ora sono tornato alla conbee2

**Reply by snakuzzo** [reply]
I firmware sono [QUI](https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware)

**Reply by xCapeITA** [reply]
o, non ufficiali, qui
https://github.com/xsp1989/zigbeeFirmware/tree/master/firmware/Zigbee3.0_Dongle-NoSigned

In ogni caso spero diventi compatibile presto

**Reply by indomus** [reply]
> > cmq continua a non trovarmi nessun dispositivo quindi direi che ho risolto poco o niente
> 
> Anche io sto litigando con questa chiavetta dopo averla presa vedendo il post su indomus. Uso Z2M e a volte pare che la chiavetta venga riconosciuta, a volte no. Come scritto dall'autore di Z2M tutti i chip EFRxxx sono in via sperimentale e il supporto è inadeguato. Viene consigliato di
> 
> 1. cambiare FW del coordinator (che non trovo. trovo solo quelli per la v1 con chip CPxxxx)
> 2. Di usare Z2M in versione EDGE (developer)
> 3. di pregare
> 
> per ora sono tornato alla conbee2

hai messo la configurazione specifica?

```
serial:
  port: /dev/ttyXXX
  adapter: ezsp
```

**Comment by simbus82** [comment]
Porto la mia esperienza se può essere utile.
A me è arrivata e l'ho installata davvero velocemente, 0 problemi.

Se la volete usare come coordinator (nella maggiorparte dei casi quindi), non serve cambiare FW perchè arriva già con EZSP v8 - 6.10.3.0 build 297.

Gli step che ho fatto:
1. Aggiornato z2m all'ultima versione 1.27.2
2. spento il servizio z2m
3. tolta dalla usb la vecchia CC2531
4. inserita la nuova Dongle-E basata su EFRxx
5. lanciato un bel `$ ls -l /dev/serial/by-id` per assicurarmi del nuovo nome della porta del device (che infatti è cambiata da `ttyUSB.CC2531-01` a `ttyACM0`)
6. **cancellato il `coordinator_backup.json`** dalla cartella `/opt/zigbee2mqtt/data`
7. **cancellato il `database.db`** dalla cartella `/opt/zigbee2mqtt/data`
8. **se non seguite i punti 6 e 7, zigbee2mqtt non vedrà mai la nuova Dongle-E!!!**
9. cambiato nel `configuration.yaml` in `/opt/zigbee2mqtt/data`  da:
```
  serial:
    port: /dev/ttyUSB.CC2531-01
```
  a:
```
  serial:
      port: /dev/ttyACM0
      adapter: ezsp
```
10. riavviato il servizio z2m
11. aspettato 2 minuti
12. controllato sul frontend z2m che fosse tutto ok
![image](https://user-images.githubusercontent.com/1375730/191133445-997c446e-dfce-48d2-9ad4-9ff58ce3b639.png)
13. Lancianto un bel Permit Join di 10 minuti
14. Fatto il pairing dei router presenti (alcuni lampade Ikea, alcuni repeater)
15. Atteso qualche minuto
16. Fatto il pairing dei dispositivi/sensori/ecc.
17. Gioito al fatto che ha mantenuto tutti i friendly name precedentemente impostati (che stanno nel configuration.yaml quindi nessuno li ha toccati).
18. Non ho riavviato nulla, né il RPi né Home Assistant.
19. 

Tutto perfettamente funzionante, con **LQI altissimi** rispetto alla vecchia CC2531!


![image](https://user-images.githubusercontent.com/1375730/191135435-0e6184bd-27cb-4b61-93eb-c3f67be027ab.png)

**Comment by fabbbio986** [comment]
Ciao a tutti,
Anche io sto avendo parecchi problemi con questo dongle e zigbee2mqtt.
Al momento ho tutta la mia rete zigbee su conbee II e deconz senza particolari problemi ma zigbee2mqtt mi ha sempre attirato per la maggior compatibilità e flessibilità.
Non ho mai trovato la voglia di riconfigurare tutta la rete ma circa una settimana fa mi ci sono messo e ho iniziato a migrare tutto usando la chiave sonoff dongle e zigbee2mqtt.
Ho seguito i passi riportati in questo [video](https://www.youtube.com/watch?v=4y_dDgo0i2g)
Non sono riuscito a venirne a capo.
Ho disinstallato e reinstallato 4-5 volte zigbee2mqtt e mosquitto broker ma ogni volta un dispositivo diverso dava problemi.
In teoria sono tutti dispostivi compatibili sull'elenco che si trova sul sito di zigbee2mqtt (deviatori vimar, lampadine philips hue, sensori di movimento aqara, lampadine ikea, ecc).
I problemi riscontrati sono legati al fatto che dopo un numero random (a volte 1 a volte 5) di dispositivi aggiunti alla rete, ogni comando inviato tramite home assistant verso dispositivi zigbee associati al dongle sonoff fa freezare zigbee2mqtt per un tempo imprecisato. Lo si vede dal fatto che i sensori di movimento non cambiano stato e che, per esempio, inviando un comando di accensione luce, questa si accende dopo 2-3, fino anche a 5, minuti dopo.
Quando succede questa situazione ogni comando da interfaccia da un errore di timeout (anche cancellare un dispositivo o attivare/disattivare la modalità di pairing cliccando sul tasto "permit join (all)").
Un altro comportamento strano mi è capitato con una lampada ikea: al primo tentativo riconosciuta ed aggiunta senza problemi, negli ultimi due tentativi è stata aggiunta ma come dispositivo non compatibile (con errore di failed interview).
Al momento sono ritornato alla configurazione con conbee II + deconz.

p.s. Guardando le risposte date dai vari utenti, mi sono soffermato sulle schermate inserite da @simbus82 . In particolare ho notato che nella versione di zigbee2mqtt affianco a 1.27.2 c'è scritto "commit:7dc48fb6".
Io ho installato la stessa versione ma compare "commit:unknow".
Potrebbe essere un problema? Cosa indica? Magari c'entra, magari no, ma è l'unica differenza che riscontro rispetto alla configurazione visibile sugli screenshot di @simbus82 .

**Reply by simbus82** [reply]
Io la cosa che sto riscontrando leggendo in rete è che chi ha problemi sembra essere chi ha Z2M in soluzioni docker, venv, ecc.
Mi sa che Zigbee2mqtt per la sua delicatezza e come deve lavorare con l'hw (usb), funziona meglio installato **puro** su linux, seguendo queste indicazioni: https://www.zigbee2mqtt.io/guide/installation/01_linux.html#installing

Quel commit credo arrivi dalla versione di Z2M aggiornata da github, forse non ti si è installato bene.

**Reply by KC-inDomus** [reply]
su docker non ho mai avuto mezzo problema, a dirla tutta. è un'analisi un po' superficiale, non corroborata da nulla, Simone.

**Reply by snakuzzo** [reply]
@simbus82 

Questo che hai appena scritto...

_Mi sa che Zigbee2mqtt per la sua delicatezza e come deve lavorare con l'hw (usb), funziona meglio installato puro su linux_

... è una cosa assolutamente NON vera.
Per favore, non diffondiamo false informazioni, grazie.

**Reply by simbus82** [reply]
Il "mi sa che" vuol dire "per me è così".
PER ME gli applicativi che su Linux vanno a lavorare sulle perifiche hw, hanno meno problemi se installati normalmente.
Ad esempio io in VENV z2m ogni tanto dava problemi di riconoscimento della seriale/usb, installato "puro" invece zero problemi.
Magari è anche colpa di z2m eh, però il metodo empirico questo mi dà come risultato. Quindi se uno ha problemi, la prima cosa da fare è eliminare i middleware.

**Reply by snakuzzo** [reply]
Mai avuto problemi di questo tipo...
Non c'entra niente l'utilizzo di docker o meno
E non c'entra niente nemmeno il venv

**Reply by fabbbio986** [reply]
> Io la cosa che sto riscontrando leggendo in rete è che chi ha problemi sembra essere chi ha Z2M in soluzioni docker, venv, ecc. Mi sa che Zigbee2mqtt per la sua delicatezza e come deve lavorare con l'hw (usb), funziona meglio installato **puro** su linux, seguendo queste indicazioni: https://www.zigbee2mqtt.io/guide/installation/01_linux.html#installing
> 
> Quel commit credo arrivi dalla versione di Z2M aggiornata da github, forse non ti si è installato bene.

Stasera ho riprovato ad installare per l'ennesima volta z2m.
Il commit:unknow compare anche nei log.
E' normale che compaiano delle righe di "assertion failed" nel log di avvio? 
Al primo avvio questo è quello che viene fuori:

```s6-rc: info: service z2m: starting
s6-rc: info: service socat: starting
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service z2m successfully started
s6-rc: info: service socat successfully started
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
[20:52:31] INFO: Socat not enabled
[20:52:33] INFO: Handing over control to Zigbee2MQTT Core ...
Zigbee2MQTT:info  2022-09-26 20:52:39: Logging to console and directory: '/config/zigbee2mqtt/log/2022-09-26.20-52-38' filename: log.txt
Zigbee2MQTT:info  2022-09-26 20:52:39: Starting Zigbee2MQTT version 1.27.2 (commit #unknown)
Zigbee2MQTT:info  2022-09-26 20:52:39: Starting zigbee-herdsman (0.14.53)
Assertion failed: Command (setConfigurationValue) returned unexpected state: [object Object]
Assertion failed: Command (setConfigurationValue) returned unexpected state: [object Object]
Assertion failed: Command (setValue) returned unexpected state: [object Object]
Assertion failed: Command (setValue) returned unexpected state: [object Object]
Zigbee2MQTT:info  2022-09-26 20:52:41: zigbee-herdsman started (resumed)
Zigbee2MQTT:info  2022-09-26 20:52:41: Coordinator firmware version: '{"meta":{"maintrel":"3 ","majorrel":"6","minorrel":"10","product":8,"revision":"6.10.3.0 build 297"},"type":"EZSP v8"}'
Zigbee2MQTT:info  2022-09-26 20:52:41: Currently 0 devices are joined:
Zigbee2MQTT:info  2022-09-26 20:52:41: Zigbee: disabling joining new devices.
Zigbee2MQTT:info  2022-09-26 20:52:41: Connecting to MQTT server at mqtt://core-mosquitto:1883
Zigbee2MQTT:info  2022-09-26 20:52:41: Connected to MQTT server
Zigbee2MQTT:info  2022-09-26 20:52:41: MQTT publish: topic 'zigbee2mqtt/bridge/state', payload 'online'
Zigbee2MQTT:info  2022-09-26 20:52:41: Started frontend on port 0.0.0.0:8099
Zigbee2MQTT:info  2022-09-26 20:52:41: MQTT publish: topic 'zigbee2mqtt/bridge/config', payload '{"commit":"unknown","coordinator":{"meta":{"maintrel":"3 ","majorrel":"6","minorrel":"10","product":8,"revision":"6.10.3.0 build 297"},"type":"EZSP v8"},"log_level":"info","network":{"channel":11,"extendedPanID":221,"panID":6754},"permit_join":false,"version":"1.27.2"}'
Zigbee2MQTT:info  2022-09-26 20:52:41: MQTT publish: topic 'zigbee2mqtt/bridge/state', payload 'online'```

**Reply by snakuzzo** [reply]
A prescindere dal fatto che sembra partito correttamente alla fine,  hai provato a dare un'occhiata tra le issue del progetto?

**Reply by snakuzzo** [reply]
Se ti riferisci a ...

```
Starting Zigbee2MQTT version 1.27.2 (commit #unknown)
```

non credo tu debba farti troppi problemi. Si riferisce alla commit su github.
Ad ogni modo... appena controllato sulla mia istanza su docker e...

```
Zigbee2MQTT:info  2022-09-26 21:52:47: Starting Zigbee2MQTT version 1.27.2 (commit #7dc48fb)
```

**Reply by fabbbio986** [reply]
si, ho provato a dare un'occhiata tra le issue ma senza successo...

Mentre per quanto riguarda le 4 righe riportanti "assertion failed" è normale che compaiano?

Potrebbe essere legato al fatto che sto provando a fare un passaggio da deconz a zigbee2mqtt graduale e quindi al momento ho 2 dongle attaccati alle usb contemporaneamente (una conbee II per deconz e una sonoff dongle "e" per z2m)?

**Reply by snakuzzo** [reply]
non credo proprio.
Quanto a quelle "assertion failed"...io non so dirtelo perché non ho quel dongle. Posso dirti che è la prima volta che vedo quei messaggi.
Vai su github e apri tu una issue...non trovi sia la cosa più semplice ?

Ad ogni modo, dal log che hai postato Z2M parte correttamente.

**Reply by fabbbio986** [reply]
Proverò ad aprire una issue sul github di z2m.
Avevo trovato questa discussione che parlava di z2m con il dongle sonoff versione "e" e mi sembrava potesse essere pertinente...

Ultima cosa: che voi sappiane per ripulire home assistant da tutte le vecchie installazioni di z2m è sufficiente disinstallare l'add on e cancellare la relativa cartella tramite vscode, oppure c'è un modo per ripulire in maniera più approfondita?

**Reply by snakuzzo** [reply]
L'addon ha una sua configurazione. Elimini la dir dove ci sono i dati di Zigbee2mqtt.
Quando lo farai poi su HA probabilmente ti ritroverai le entità precedentemente create come non più disponibili...ma basta rimuoverle tutte.
Fine

**Comment by Mcpidess** [comment]
Mi ha dato problemi con Rpi 4 64 bit. Per ora nessun problema con 32 bit

**Comment by Eros-ITA** [comment]
A me si è magicamente risolto tutto a dopo gli aggiornamenti

**Reply by fabbbio986** [reply]
A che aggiornamenti ti riferisci? Home assistant, zigbee2mqtt o altro?

**Comment by marcoferroha** [comment]
non riesco a farla funzionare ricevo sempre:
Using '/app/data' as data directory
Zigbee2MQTT:info  2022-10-24 10:06:41: Logging to console and directory: '/var/log' filename: log.txt
Zigbee2MQTT:info  2022-10-24 10:06:41: Starting Zigbee2MQTT version 1.28.0 (commit #03ba647)
Zigbee2MQTT:info  2022-10-24 10:06:41: Starting zigbee-herdsman (0.14.62)
Zigbee2MQTT:error 2022-10-24 10:06:45: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2022-10-24 10:06:45: Failed to start zigbee
Zigbee2MQTT:error 2022-10-24 10:06:45: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2022-10-24 10:06:45: Exiting...
Zigbee2MQTT:error 2022-10-24 10:06:45: Error: Failure send version:{"type":"Buffer","data":[1,0,1,0,0,8]}
    at /app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/ezsp.ts:507:23
    at Queue.executeNext (/app/node_modules/zigbee-herdsman/src/utils/queue.ts:32:32)
Using '/app/data' as data directory


Qualche suggerimento?

**Reply by snakuzzo** [reply]
Controlla bene tutto quanto. Hai specificato la porta corretta ed il parametro che indica che è una chiavetta con chip diverso?

**Comment by marcoferroha** [comment]
/dev/bus/usb/001/001 - Linux_5.15.73-v8+_dwc_otg_hcd_DWC_OTG_Controller_3f980000.usb
/dev/bus/usb/001/007 - SIGMACHIP_USB_Keyboard
/dev/input/event3 - SIGMACHIP_USB_Keyboard
/dev/input/event2 - SIGMACHIP_USB_Keyboard
/dev/hidraw2 - 
/dev/input/event1 - SIGMACHIP_USB_Keyboard
/dev/hidraw1 - 
/dev/ttyACM0 - ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20220810165023
/dev/bus/usb/001/005 - ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20220810165023
/dev/bus/usb/001/003 - 0424_2514
/dev/bus/usb/001/006 - Microsoft_Microsoft_Basic_Optical_Mouse_v2.0
/dev/input/mouse0 - Microsoft_Microsoft_Basic_Optical_Mouse_v2.0
/dev/input/event0 - Microsoft_Microsoft_Basic_Optical_Mouse_v2.0
/dev/hidraw0 - 
/dev/bus/usb/001/008 - 0424_7800
/dev/bus/usb/001/002 - 0424_2514
/dev/bus/usb/001/009 - dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2258172
/dev/ttyACM1 - dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2258172

la ConbeeII funziona perfettamente 

Mosquitto anche (configurato e raggiunto da Homeassistant)

Questo è il config di zigbee2mqtt:

homeassistant: true
# Join alla rete MQTT disabilitato di default
permit_join: true
# MQTT settings
mqtt:
  # MQTT base topic per i messaggi di zigbee2mqtt
  base_topic: zigbee2mqtt
  # MQTT server - URL (indirizzo LAN)
# server: 'mqtt://IP_DEL_BROKER_MQTT'
  server: mqtt://192.168.0.140
  # MQTT server - Parametri di autenticazione (decommentare e popolare correttamente se si utilizzano credenziali per la connessione al broker MQTT
  user: (nascosto per ovvi motivi)
  password: (nascosta per ovvi motivi)
# version: 5
  include_device_information: true
serial:
  # Porta su cui è mappata l'antennza zigbee
  port: /dev/ttyACM0
  adapter: ezsp
advanced:
  # Directory in cui verrà salvato il file di log
  log_directory: /var/log
# experimental: {}
frontend:
  # porta su cui verrà esposta la webGUI
  port: 8080
ota:
  update_check_interval: 10
disable_automatic_update_check: false

**Reply by snakuzzo** [reply]
Hai seguito il link che ti appariva nei log?
Hai controllato che il firmware sia effettivamente presente sulla chiavetta? (Segui la guida che abbiamo scritto per il flash)

**Reply by xCapeITA** [reply]
Da quello che ho capito con l'esperienza..
1- usa come serial il /dev/ttyACM0 e non la stringa con id
2 - Non mantenere alcun dato della rete precedente con la Conbee (soprattutto la network key se impostata) e resetta ogni device. Z2M ti avvertirà che il dispositivo si è disconnesso dalla rete e potrai riconnetterlo.
3 - quando il countdown permit join si esaurisce RIAVVIA Z2M oppure non riuscirai a fare pairing di altri dispositivi (anche se avrai il countdown attivo)

Solo così sono riuscito ad avviare tutto e farlo funzionare.
Nota a margine.. non è che (per il momento) sia cambiato granché rispetto alla conbee² .. magari in futuro si apprezzerà meglio (speriamo)

**Comment by marcoferroha** [comment]
Tutto controllato il dispositivo l'ho installato da solo in un altra 
macchina e funziona perfettamente

In data 24 ottobre 2022 12:02:29 Diego Ambrosanio 
***@***.***> ha scritto:

**Reply by snakuzzo** [reply]
Quindi ti sarai perso qualcosa

**Comment by marcoferroha** [comment]
Non so, docker è su perché tutto il resto funziona, Ha funziona, mosquito 
funziona, l'unico è zigbee2mqtt...

In data 24 ottobre 2022 14:02:54 Diego Ambrosanio 
***@***.***> ha scritto:

**Reply by snakuzzo** [reply]
Per favore non fare più reply direttamente dalla mail creando di conseguenza ogni volta nuovi thread...che poi non si capisce più una fava.

**Reply by snakuzzo** [reply]
Controlla quanto indicato al link che ti indica nel log

**Comment by marcoferroha** [comment]
Anche Deconz funziona

In data 24 ottobre 2022 14:02:54 Diego Ambrosanio 
***@***.***> ha scritto:

**Comment by marcoferroha** [comment]


**Reply by snakuzzo** [reply]
Stai continuando a fare reply dalla mail !!!
